src/Forms/ConfirmedPasswordField.php

```diff
@@ -562,9 +598,18 @@ public function saveInto(DataObjectInterface $record)
             return;
         }
 
-        if (!($this->canBeEmpty && !$this->value)) {
-            parent::saveInto($record);
+        // Create a random password for Member if password is blank
+        if (!$this->value
+            && $this->canBeEmpty
+            && $this->generateRandomPasswordOnEmpty
+        ) {
+            if (!is_callable($this->randomPasswordCallback)) {
+                throw new LogicException('randomPasswordCallback must be callable');
+            }
+            $this->value = call_user_func($this->randomPasswordCallback, ['maxLength' => $this->maxLength]);
         }
+
+        parent::saveInto($record);
```

Comment:
Shouldn't we only be saving in the following scenarios?
1. A value is set
2. No value is set, but we are allowed to be empty

I.e. `if (!$this->value && !$this->canBeEmpty)` - in that scenario, we shouldn't save, right?

Reply:
Correct - logic for that is already in ConfirmedPasswordField::validate()

Reply:
This method is public and can be called without first calling validate. Changing this behaviour now is a breaking change. Please restore the original conditional saving behaviour.

Reply:
Updated